Create ui-c8y-1023-64-0-as-a-user-i-want-the-data-point-list-widget-to.md (#4475)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1023-64-0-as-a-user-i-want-the-data-point-list-widget-to.md

---
date: ""

> title: Data Point List widget improvements

Reviewer: ```suggestion
title: Data point list widget improvements
```

> title: Data point list widget improvements

Reviewer 2: ```suggestion
title: Enhanced Data point list widget
```

product_area: Application enablement & solutions
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-YdSEScrEC
    label: Cockpit
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-65026
version: 1023.64.0
---

> The Data Point List widget has been enhanced with the following improvements:

Reviewer: ```suggestion
The "Data point list" widget has been enhanced with the following improvements:
```

> The "Data point list" widget has been enhanced with the following improvements:

Reviewer 2: ```suggestion
The "Data point list" widget has been migrated to Angular and enhanced to improve usability and performance, providing a modern, consistent foundation with the current UI framework. Updates include:
```

> - Decimal place configuration: Specify decimal places for numerical values, with a default of 2 to align with other Cumulocity IoT widgets.

Reviewer: ```suggestion
- **Decimal place configuration**: Specify decimal places for numerical values, with a default of 2 to align with other {{< product-c8y-iot >}} IoT widgets.
```

> - Global time context integration: The widget now responds to dashboard time context changes, automatically updating to show the last measurement value for the selected time range.

Reviewer: ```suggestion
- **Global time context integration**: The widget now responds to dashboard time context changes, automatically updating to show the last measurement value for the selected time range.
```

> - Asset icon display: Asset icons are now shown alongside asset names for easier visual identification.

Reviewer: ```suggestion
- **Asset icon display**: Asset icons are now shown alongside asset names for easier visual identification.
```

> - Export capabilities: Export displayed data in CSV and XLSX formats for external use.

Reviewer: ```suggestion
- **Export capabilities**: Export displayed data in CSV and XLSX formats for external use.
```

> - Field arrangement: Customize the display order of fields within the widget.

Reviewer: ```suggestion
- **Field arrangement**: Customize the display order of fields within the widget.
```

> - Performance optimization: Improved rendering performance through virtual scrolling — only visible rows are rendered, ensuring smooth operation even with large datasets.

Reviewer: ```suggestion
- **Performance optimization**: Improved rendering performance through virtual scrolling — only visible rows are rendered, ensuring smooth operation even with large datasets.
```

---
date: ""

> title: As a user, I want the Data Point List widget to be enhanced so I can view more comprehensive and reliable data (#10193) [GRAFT][release/cd] (#11624)

Reviewer 3: The Claude AI suggested to use the following title:
```suggestion
title: Data Point List widget improvements
```

product_area: Application enablement & solutions
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-YdSEScrEC
    label: Cockpit
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-65026
version: 1023.64.0
---
# Backport

This will backport the following commits from `develop` to `release/cd`:
- [feat(cockpit): [MTM-65026] As a user, I want the Data Point List
widget to be enhanced so I can view more comprehensive and reliable data
(#10193)](https://github.com/Cumulocity-IoT/cumulocity-ui/pull/10193)

<!--- Backport version: 9.5.1 -->

### Questions ?
Please refer to the [Backport tool
documentation](https://github.com/sorenlouv/backport)

[MTM-65026]:
https://cumulocity.atlassian.net/browse/MTM-65026?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Co-authored-by: Lukasz Janusz <114157358+luja-c8y@users.noreply.github.com>
Co-authored-by: Lukasz Janusz <lukasz.janusz@cumuloicty.com>
Co-authored-by: Dawid Janusz <dawid.janusz@cumulocity.com>

> Co-authored-by: Carlos Ceia <carlos.ceia@cumulocity.com>

Reviewer 3: The Claude AI suggested to use the following body:
```suggestion
The Data Point List widget has been enhanced with the following improvements:

- Decimal place configuration: Specify decimal places for numerical values, with a default of 2 to align with other Cumulocity IoT widgets.
- Global time context integration: The widget now responds to dashboard time context changes, automatically updating to show the last measurement value for the selected time range.
- Asset icon display: Asset icons are now shown alongside asset names for easier visual identification.
- Export capabilities: Export displayed data in CSV and XLSX formats for external use.
- Field arrangement: Customize the display order of fields within the widget.
- Performance optimization: Improved rendering performance through virtual scrolling — only visible rows are rendered, ensuring smooth operation even with large datasets.
```
